seacrowd/sea_datasets/limesoda/limesoda.py

```diff
@@ -0,0 +1,174 @@
+
+        if self.config.schema == "source":
+            if self.config.subset_id == "limesoda_raw":
+                features = datasets.Features(
+                    {
+                        "id": datasets.Value("string"),
+                        "title": datasets.Value("string"),
+                        "detail": datasets.Sequence(datasets.Value("string")),
+                        "title_token_tags": datasets.Value("string"),
+                        "detail_token_tags": datasets.Sequence(datasets.Value("string")),
+                        "document_tag": datasets.Value("string"),
+                    }
+                )
+            else:
+                features = datasets.Features({"id": datasets.Value("string"), "text": datasets.Value("string"), "document_tag": datasets.Value("string")})
+        elif self.config.schema == "seacrowd_text":
+            features = schemas.text_features(["Fact News", "Fake News", "Undefined"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        path_dict = dl_manager.download_and_extract(_URLS)
+        if self.config.subset_id == "limesoda_raw":
+            raw_path = path_dict["raw"]
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # Whatever you put in gen_kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": raw_path,
+                    },
+                ),
+            ]
+        elif self.config.subset_id == "limesoda_split":
+            train_path, val_path, test_path = path_dict["split"]["train"], path_dict["split"]["val"], path_dict["split"]["test"]
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # Whatever you put in gen_kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": train_path,
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={
+                        "filepath": test_path,
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    gen_kwargs={
+                        "filepath": val_path,
+                    },
+                ),
+            ]
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        with open(filepath, "r") as f:
+            df = [json.loads(line) for line in f.readlines()]
```

Comment:
I suggest changing the variable name `df` into something more informative. I usually see this in the context of a pandas DataFrame, but I don't see any use of that library here.